doc/source/release/1.12.0-notes.rst

```diff
@@ -49,42 +84,118 @@ New features
 
 `scipy.optimize` improvements
 =============================
-
+- `scipy.optimize.nnls` is rewritten in Python and now implements the so-called
+  fnnls or fast nnls.
+- The result object of `scipy.optimize.root` and `scipy.optimize.root_scalar`
+  now reports the method used.
+- The ``callback`` method of `scipy.optimize.differential_evolution` can now be
+  passed more detailed information via the ``intermediate_results`` keyword
+  parameter. Also, the evolution ``strategy`` now accepts a callable for
+  additional customization.
+- ``minimize`` method ``Newton-CG`` has been made slightly more efficient.
+- ``minimize`` method ``BFGS`` now accepts an initial estimate for the inverse
+  of the Hessian, which allows for more efficient workflows in some
+  circumstances. The new parameter is ``hess_inv0``.
+- ``minimize`` methods ``CG``, ``Newton-CG``, and ``BFGS`` now accept parameters
+  ``c1`` and ``c2``, allowing specification of the Armijo and curvature rule
+  parameters, respectively.
+- ``curve_fit`` performance has improved due to more efficient memoization
+  of the callable function.
 
 `scipy.signal` improvements
 ===========================
-
```

Comment:
I think, due to its considerable length alone 😁, PR #17408 deserves a mention.  A possible wording could look like this:

```ReST
- The class ``ShortTimeFFT`` provides a new more versatile implementation of the
  short-time Fourier transform (STFT), its inverse (ISTFT) as well as the (cross-)
  spectrogram. It utilizes an improved algorithm for calculating the ISTFT based on
  dual windows and provides more fine-grained control of the parametrization especially
  in regard to scaling and phase-shift. Functionality was implemented to ease
  working with signal and STFT chunks. A section has been added to the "SciPy User Guide"
  providing algorithmic details. The functions ``stft``, ``istft`` and ``spectrogram``
  have been marked as legacy.
```